src/ringmesh/io/geomodel/io_gm.cpp

```diff
@@ -341,14 +341,25 @@ namespace
             return false;
         }
 
-        void load_file() final
+        std::string generate_folder_name() const
         {
-            const std::string directory_to_unzip{
+            std::string directory_to_unzip{
                 GEO::FileSystem::normalized_path(
                     GEO::FileSystem::dir_name( this->filename_ ) )
                 + "/" + std::to_string(
                             std::hash< std::string >()( this->filename_ ) )
             };
+            index_t unique_id{ 0 };
+            while( GEO::FileSystem::is_directory( directory_to_unzip ) )
```

Comment:
Ok for the idea but this is not thread safe at all... Why if two threads run the is_directory function at the same time ? or if they enter inside the while loop at the same time? 

Look at double-check locking, I think this will be a possible way to solve this issue.